chapter2/tests/t19/terraform/base/variables.tf

```diff
@@ -0,0 +1,110 @@
+/* 
+  ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+  │ nginx configuration variables                                                                                    │
+  └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+ */
+
+variable "nginx" {
+  type = object({
+    image           = string
+    container_name  = optional(string)
+    container_ports = optional(map(string))
+    container_volumes = list(object({
+      host      = string
+      container = string
+    }))
+    keep_locally = bool
+  })
+  default = {
+    image             = "nginx:latest"
+    keep_locally      = false
+    container_volumes = []
+  }
+  validation {
+    condition = (
+      can(var.nginx.container_ports) &&
+      var.nginx.container_ports["internal"] < "1000" &&
+      var.nginx.container_ports["external"] >= "8000"
+    )
+    error_message = "Container internal port should be less 1000 and external above or equal to 8000"
+  }
+  validation {
+    condition     = can(regex("^(saritasa-devops-camps-2023-).*", var.nginx.container_name))
+    error_message = "Container name should be prefixed with saritasa-devops-camps-2023-"
+  }
+}
+
+variable "container_volumes" {
+  type = list(object({
+    host      = string
+    container = string
+  }))
+  default = []
+}
+
+/* 
+  ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+  │ redis configuration variables                                                                                    │
+  └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+ */
+
+
+variable "use_redis" {
+  description = "Do you need to speed up the perf using redis?"
+  type        = bool
+  default     = false
+}
+
+variable "redis" {
+  type = object({
+    image           = string
+    container_name  = optional(string)
+    container_ports = optional(map(string))
+    keep_locally    = bool
+  })
+  default = {
```

Comment:
prefer defaults in infra.auto.tfvars, not in variables.tf as it makes it longer

Reply:
> prefer defaults in infra.auto.tfvars, not in variables.tf as it makes it longer

@dmitry-mightydevops 
Done